pydm/widgets/slider.py

```diff
@@ -217,22 +234,42 @@ def apply_step_size_menu_changes(self):
         """
         Method which attempts to set the user imputed data from the slider parameters menu.
         """
+        # check
+        try:
+            slider_value = float(self.slider_parameters_menu_input_widgets[0].text())
+
+            if slider_value < self.minimum or slider_value > self.maximum:
+                raise ValueError
+            if slider_value != self.value:
+                self.remap_flag = True
+                self.value_changed(slider_value)
+                self.send_value_signal[float].emit(self.value)
+        except ValueError:
+            logger.error("the given value is not a valid type or outside of the slider range")
+
         try:
+            self.remap_flag = True
+            # checks if input can be converted to a float if not connect to a new channel
             new_step_size = float(self.slider_parameters_menu_input_widgets[1].text())
             new_step_size_scaled = new_step_size * float(self.slider_parameters_menu_input_widgets[2].currentText())
             if new_step_size_scaled > 0:
-                self.step_size = new_step_size_scaled
+                # if new step_size scaled greater than zero we want to
+                # disconnect the step size channel and reset step size
 
-                if self.step_size_channel is not None:
-                    self.step_size_channel.disconnect()
+                if self.step_size_channel_pv is not None:
+                    self.step_size_channel_pv.disconnect()
+                    self.step_size_channel_pv = None
                     self.step_size_channel = None
+                self.step_size = new_step_size_scaled
             else:
+                # calc step size and update?
```

Comment:
`# calc step size and update?`
comment can be removed, this looks like an error condition and is find to just log it